pkgs/pkgs-lib/formats.nix

```
@@ -473,4 +473,56 @@ rec {
     '') {};
   };
 
+  envvars = {
+    export ? false
+  , listToStringSep ? ","
+  , listToStringFn ? lib.concatStringsSep listToStringSep
+  , boolToStringFn ? lib.boolToString
+  }: {
+    type = with lib.types; let
+      unescapedType = addCheck (attrsOf str) (attrs: (lib.attrNames attrs) == [ "_type" "value" ] && attrs._type == "unescaped");
```

I'd rather not rely on `oneOf`, `addCheck` and `attrsOf` to make sure that `.value` is a string. How about this instead:

```suggestion
      unescapedType =
        addCheck raw (attrs:
          attrs ? _type
          && attrs._type == "unescaped"
          && attrs ? value
          && isString attrs.value
        );
```

---

As well as `isAttrs attrs` then maybe?

---

Could be done too, but shouldn't be necessary because `?` fails on non-attrs too.